test/gc/transaction_level_gc_manager_test.cpp

```diff
@@ -486,5 +521,192 @@ TEST_F(TransactionLevelGCManagerTests, ImmutabilityTest) {
   txn_manager.CommitTransaction(txn);
 }
 
+// check mem -> insert 100k -> check mem -> delete all -> check mem
+TEST_F(TransactionLevelGCManagerTests, FreeTileGroupsTest) {
+
+  auto &epoch_manager = concurrency::EpochManagerFactory::GetInstance();
+  epoch_manager.Reset(1);
+
+  std::vector<std::unique_ptr<std::thread>> gc_threads;
+
+  gc::GCManagerFactory::Configure(1);
+  auto &gc_manager = gc::TransactionLevelGCManager::GetInstance();
+  gc_manager.Reset();
+
+  auto storage_manager = storage::StorageManager::GetInstance();
+  // create database
+  auto database = TestingExecutorUtil::InitializeDatabase("FreeTileGroupsDB");
+  oid_t db_id = database->GetOid();
+  EXPECT_TRUE(storage_manager->HasDatabase(db_id));
+
+  // create a table with only one key
+  const int num_key = 0;
+  size_t tuples_per_tilegroup = 2;
+
+  std::unique_ptr<storage::DataTable> table(TestingTransactionUtil::CreateTable(
+      num_key, "TABLE1", db_id, INVALID_OID, 1234, true, tuples_per_tilegroup));
+
+  auto &manager = catalog::Manager::GetInstance();
+  size_t tile_group_count_after_init = manager.GetNumLiveTileGroups();
+  LOG_DEBUG("tile_group_count_after_init: %zu\n", tile_group_count_after_init);
+
+  auto current_eid = epoch_manager.GetCurrentEpochId();
+
+  //  int round = 1;
+  for(int round = 1; round <= 3; round++) {
+
+    LOG_DEBUG("Round: %d\n", round);
+
+    epoch_manager.SetCurrentEpochId(++current_eid);
+    //===========================
+    // insert tuples here.
+    //===========================
+    size_t num_inserts = 100;
+    auto insert_result = BulkInsertTuples(table.get(), num_inserts);
+    EXPECT_EQ(ResultType::SUCCESS, insert_result);
+
+    // capture memory usage
+    size_t tile_group_count_after_insert = manager.GetNumLiveTileGroups();
+    LOG_DEBUG("Round %d: tile_group_count_after_insert: %zu", round, tile_group_count_after_insert);
+
+    epoch_manager.SetCurrentEpochId(++current_eid);
+    //===========================
+    // delete the tuples.
+    //===========================
+    auto delete_result = BulkDeleteTuples(table.get(), num_inserts);
+    EXPECT_EQ(ResultType::SUCCESS, delete_result);
+
+    size_t tile_group_count_after_delete = manager.GetNumLiveTileGroups();
+    LOG_DEBUG("Round %d: tile_group_count_after_delete: %zu", round, tile_group_count_after_delete);
+
```

Comment:
It's better to check the value of  tile_group_count_after_delete here, using EXPECT function.